drivers/dma/dma_silabs_siwx917.c

```diff
@@ -0,0 +1,480 @@
+		return -EINVAL;
+	}
+	/* Validate burst length */
+	if ((siwx917_dma_burst_length(config->source_burst_length >> 3) < 0) ||
+	    (siwx917_dma_burst_length(config->dest_burst_length >> 3) < 0)) {
+		return -EINVAL;
+	}
+	channel_control.srcSize = siwx917_dma_data_width(config->source_data_size);
+	channel_control.dstSize = siwx917_dma_data_width(config->dest_data_size);
+	/* Validate the addr increment value */
+	if ((siwx917_dma_addr_adjustment(config->head_block->source_addr_adj) < 0) ||
+	    (siwx917_dma_addr_adjustment(config->head_block->dest_addr_adj) < 0)) {
+		return -EINVAL;
+	}
+	/* Update source and destination addr increment values */
+	if (siwx917_dma_addr_adjustment(config->head_block->source_addr_adj) == 0) {
+		channel_control.srcInc = channel_control.srcSize;
+	} else {
+		channel_control.srcInc = UDMA_SRC_INC_NONE;
+	}
+	if (siwx917_dma_addr_adjustment(config->head_block->dest_addr_adj) == 0) {
+		channel_control.dstInc = channel_control.dstSize;
+	} else {
+		channel_control.dstInc = UDMA_DST_INC_NONE;
+	}
+	/* Configure dma channel for transfer */
+	status = (int)UDMAx_ChannelConfigure(&UDMA_Resources, (uint8_t)rsi_channel,
+					     (uint32_t)(config->head_block->source_address),
+					     (uint32_t)(config->head_block->dest_address),
+					     config->head_block->block_size, channel_control,
+					     &channel_config, NULL, channel_info, udma_handle);
+	return status;
+}
+
+/* Function to configure UDMA channel for transfer */
+static int dma_siwx917_configure(const struct device *dev, uint32_t channel,
+				 struct dma_config *config)
+{
+	const struct dma_siwx917_config *cfg = dev->config;
+	struct dma_siwx917_data *data = dev->data;
+	uint32_t rsi_channel = channel - 1;
+	int status;
+	RSI_UDMA_HANDLE_T udma_handle = (RSI_UDMA_HANDLE_T)data->dma_rom_buff;
+
+	/* Expecting a fixed channel number between 1-32 for udma0 and 1-12 for udma1 */
+	if ((channel > cfg->channels) || (channel == 0)) {
+		return -EINVAL;
+	}
+
+	/* Disable the channel before configuring */
+	if (RSI_UDMA_ChannelDisable(udma_handle, rsi_channel) != 0) {
+		return -EIO;
+	}
+
+	/* Validate the priority */
+	if ((config->channel_priority != DMA_CH_PRIORITY_LOW) &&
+	    (config->channel_priority != DMA_CH_PRIORITY_HIGH)) {
+		return -EINVAL;
+	}
+
+	/* Configure dma channel for transfer */
+	status = dma_channel_config(dev, udma_handle, rsi_channel, config, data->udma_channel_info);
+	if (status != 0) {
+		return -ECANCELED;
+	}
+	return 0;
+}
+
+/* Function to reload UDMA channel for new transfer */
+static int dma_siwx917_reload(const struct device *dev, uint32_t channel, uint32_t src,
+			      uint32_t dst, size_t size)
+{
+	const struct dma_siwx917_config *cfg = dev->config;
+	struct dma_siwx917_data *data = dev->data;
+	uint32_t rsi_channel = channel - 1;
+	RSI_UDMA_DESC_T *UDMA_Table;
+	uint32_t desc_src_addr;
+	uint32_t desc_dst_addr;
+	RSI_UDMA_HANDLE_T udma_handle = (RSI_UDMA_HANDLE_T)data->dma_rom_buff;
```

Comment:
It seems to forgot to make the change.

BTW, to make things clearer, I suggest to explicitly use `void *` rather than `RSI_UDMA_HANDLE_T`.

(also apply this change to the other references of `dma_rom_buff`)